src/main/java/com/mineboundteam/minebound/magic/UtilitySpells/TelekineticUtilitySpell.java

```diff
@@ -0,0 +1,116 @@
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.ForgeConfigSpec.Builder;
+import net.minecraftforge.common.ForgeConfigSpec.IntValue;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.config.ModConfigEvent;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.List;
+
+
+@Mod.EventBusSubscriber(modid = MineBound.MOD_ID)
+public class TelekineticUtilitySpell extends PassiveSpellItem {
+
+    private final int manaCost;
+
+    static HashMap<ArmorTier, Double> manaCostReductions = new HashMap<>() {{
+        put(ArmorTier.SUIT, 0.20);
+        put(ArmorTier.SYNERGY, 0.35);
+        put(ArmorTier.SINGULARITY, 0.5);
+    }};
+
+    public TelekineticUtilitySpell(Properties properties, TelekineticUtilitySpellConfig config) {
+        super(properties, config.LEVEL);
+
+        manaCost = config.MANA_COST.get();
+    }
+
+    @SubscribeEvent
+    public static void onPlayerTick(TickEvent.PlayerTickEvent event) {
+        if (event.side.isServer() && event.phase == TickEvent.Phase.START && !event.player.isCreative()) {
+            Player player = event.player;
+            List<TelekineticUtilitySpell> equippedTelekineticUtilitySpells = getEquippedSpellsOfType(TelekineticUtilitySpell.class, event.player);
+
+            // Set if player should still have flight on every tick
+            player.getAbilities().mayfly = equippedTelekineticUtilitySpells.size() > 0;
+            if (!player.getAbilities().mayfly) {
+                player.getAbilities().flying = false;
+            }
+            player.onUpdateAbilities();
+
+            // Calculate mana cost only once per second (every 20 ticks) if player is flying
+            if (player.level.getDayTime() % 20 == 0 && equippedTelekineticUtilitySpells.size() > 0 && player.getAbilities().flying) {
+                TelekineticUtilitySpell highestLevelSpell = equippedTelekineticUtilitySpells.get(0);
+                for (TelekineticUtilitySpell equippedTelekineticUtilitySpell : equippedTelekineticUtilitySpells.subList(1, equippedTelekineticUtilitySpells.size())) {
+                    if (equippedTelekineticUtilitySpell.level.getValue() > highestLevelSpell.level.getValue()) {
+                        highestLevelSpell = equippedTelekineticUtilitySpell;
+                    }
+                }
+                equippedTelekineticUtilitySpells.remove(highestLevelSpell);
+                int reducedManaCost = highestLevelSpell.manaCost;
+                for (TelekineticUtilitySpell equippedTelekineticUtilitySpell : equippedTelekineticUtilitySpells) {
+                    reducedManaCost *= 1 - manaCostReductions.get(equippedTelekineticUtilitySpell.level);
+                }
+                // Will cost as little as 1 mana per second
+                highestLevelSpell.reduceMana(Math.max(reducedManaCost, 1), player);
```

Comment:
let it cost zero; if they spec their whole build around flying they'll basically be defenseless. we can rebalance the values for the reductions to make it hard enough to do this that it'll need to be a conscious choice

Reply:
Done.